config/routes.rb

```diff
@@ -2629,11 +2629,57 @@
   apiver_regex = /v[\d]+(\.[\da-zA-Z]+)*(\-[\da-zA-Z]+)?/
 
   match '/api/*path'   => 'api#handle_options_request', :via => [:options]
-  get '/api(/:version)'           => 'api#show',    :format => 'json', :version => apiver_regex
+  get '/api(/:version)'           => 'api#show_entrypoint',    :format => 'json', :version => apiver_regex
+
+  def action_for(verb, collection)
+    prefix = {
+      :get => "show",
+      :post => "update",
+      :put => "update",
+      :patch => "update",
+      :delete => "destroy"
+    }[verb]
+
+    if ApiController.method_defined?("#{prefix}_#{collection}")
```

Comment:
I think this should be limited to public methods (if this method doesn't do so already)